docs/using-modules.rst

```diff
@@ -0,0 +1,192 @@
+    import ownable as ow         # accessible as `ow`
+    from . import ownable        # accessible as `ownable`
+    from . import ownable as ow  # accessible as `ow`
+
+When importing using the ``as`` keyword, the module will be referred to by its alias in the rest of the contract.
+
+The ``_times_two()`` helper function in the above module can be immediately used without any further work.
+
+.. code-block:: vyper
+
+    import ownable as helper
+    @external
+    def my_function(x: uint256) -> uint256:
+        return helper._times_two(x)
+
+The other functions cannot be used yet, because they touch the ``ownable`` module's state. There are two ways to declare a module so that its state can be used.
+
+Initializing a module
+=====================
+
+In order to use a module's state, it must be "initialized". A module can be initialized with the ``initializes`` keyword. This declares the module's location in the contract's :ref:`Storage Layout <compiler-storage-layout>`. It also creates a requirement to invoke the module's :ref:`__init__() function <init-function>`, if it has one. This is a well-formedness requirement, since it does not make sense to access a module's state unless its ``__init__()`` function has been called.
+
+.. code-block:: vyper
+
+    import ownable
+
+    initializes: ownable
+
+    @deploy
+    def __init__():
+        ownable.__init__()
+
+    @external
+    def my_access_controlled_function():
+        ownable._check_owner()  # reverts unless msg.sender == ownable.owner
+
+        ... # do things that only the owner can do
+
+It is a compile-time error to invoke a module's ``__init__()`` function more than once!
+
+A module's state can be directly accessed just by prefixing the name of a variable with the module's alias, like follows:
+
+.. code-block:: vyper
+
+    @external
+    def get_owner() -> address:
+        return ownable.owner
+
+
+The ``uses`` statement
+======================
+
+Another way of using a contract's state without directly initializing it is to use the ``uses`` keyword. This is a more advanced usage which is expected to be mostly utilized by library designers. The ``uses`` statement allows a module to use another module's state but defer its initialization to another module in the compilation tree (most likely a user of the library in question).
+
+This is best illustrated with an example:
+
+.. code-block:: vyper
+
+    # ownable_2step.vy
+    import ownable
+
+    uses: ownable
+
+    # does not export ownable.transfer_ownership!
+
+    pending_owner: address  # the pending owner in the 2-step transfer process
+
+    @deploy
+    def __init__():
+        self.pending_owner = empty(address)
+
+    @external
+    def begin_transfer(new_owner: address):
+        ownable._check_owner()
+
+        self.pending_owner = new_owner
+
+    @external
+    def accept_transfer():
+        assert msg.sender == self.pending_owner
+
+        self.owner = new_owner
```

Comment:
Is this correct?

Should it be:

```suggestion
        ownable.owner = new_owner
```

Or does the `uses` statement implicitly elide storage of an import directly into itself?

Reply:
oops, mistake